unfoldingWord/translationCore
# Local project import Implementation

### This pull request addresses:

- Created an action to open a file/browse dialog to access local project.
- .tstudio project unzipping
- Local project loading 
- Appropriate error handling and error alerts.

### How to test this pull request:
- Use this files to test importing **.tstudio** projects [tstudio_projects.zip](https://github.com/unfoldingWord-dev/translationCore/files/1129249/tstudio_projects.zip)
- Go to the ** Projects Cards ** & click on the FAB and select **Import Local Project**
- Once the file browser opens then select a file that you know is not compatible with **tC**.
    **(You should get the appropriate error message for trying to import a noncompatible file.)** 
- Try importing a folder that contains a **manifest.json** file but that is not a valid project manifest.
    **(You should get the appropriate error message for trying to import a noncompatible file.)** 
- Try importing a .tstudio file. **(Project should load fine)**
- Try importing a (fresh and used)  valid project from a directory other than the **~./translationCore** directory.  **(Project should load fine)**


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/unfoldingword-dev/translationcore/2114)
